js/src/toast.js

```diff
@@ -176,8 +182,53 @@ class Toast extends BaseComponent {
     return config
   }
 
+  _maybeScheduleHide() {
+    if (!this._config.autohide) {
+      return
+    }
+
+    if (this._activeMouseInteraction || this._activeKeyboardInteraction) {
+      return
+    }
+
+    this._timeout = setTimeout(() => {
+      this.hide()
+    }, this._config.delay)
+  }
+
+  _onInteraction(event, interacting) {
+    switch (event.type) {
+      case 'mouseover':
+      case 'mouseout':
+        this._activeMouseInteraction = interacting
+        break
+      case 'focusin':
+      case 'focusout':
+        this._activeKeyboardInteraction = interacting
+        break
+      default:
+        break
+    }
+
+    if (interacting) {
+      this._clearTimeout()
+      return
+    }
+
+    const nextElement = event.relatedTarget
```

Comment:
this is to avoid multiple triggering events?

Reply:
This is an optimization, if this conditional is removed everything would still work fine. 

At this point in the event handler, we know that the interaction is a "leaving" one. Either `mouseout` or `focusout`. While we could just call `this._maybeScheduleHide()`, why bother if we **know** any timer it might set would be cancelled almost immediately? By checking whether or not we are about to receive an "entering" interaction (`mouseover` or `focusin`) we can save a round trip of creating, and then immediately clearing a timer.

Reply:
The `nextElement` I mean , but I guess is to  check the [previous ](https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/relatedTarget#javascript)target . Right?

Reply:
I think we are talking about the same thing, I might just be unclear with my explanation. Let me try again - `nextElement` is the most meaningful term to describe `event.relatedTarget` in this case. At this point in the conditional the event is a "leaving" event. Either `mouseout` or `focusout`. The `relatedTarget` of a leaving event is the element that the users interaction is moving to _next_, just like the mdn link you referenced outlines. So in the case of a `mouseout`, `nextElement` assigned the element that the cursor will move to next, and in the case of `focusout`, `nextElement` is assigned to the element the the focus will move to next.

So, the conditional referencing the `nextElement` says "if the user is just moving to another element in the toast, don't even bother trying to schedule a hide".

Reply:
I get it. thank you.